operator/api/pulumi/v1/program_types.go

```diff
@@ -160,16 +160,35 @@ type Getter struct {
 	State map[string]Expression `json:"state,omitempty"`
 }
 
+// StackStatus defines the observed state of Program.
+type ProgramStatus struct {
+	// ObservedGeneration is the last observed generation of the Program
+	// object.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// LastResyncTime contains a timestamp for the last time a resync of the stack took place.
+	// +optional
+	LastResyncTime metav1.Time `json:"lastResyncTime,omitempty"`
+
+	// Artifact represents the last succesful artifact generated by program reconciliation.
+	// +optional
+	Artifact *Artifact `json:"artifact,omitempty"`
+}
+
 // +kubebuilder:object:root=true
 // +kubebuilder:storageversion
+// +kubebuilder:subresource:status
 // +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.conditions[?(@.type==\"Ready\")].message",description=""
```

Comment:
The Program doesn't have a Ready condition, assumedly you had planned to before going "on-demand".  Maybe it still should have a ready condition, if you would want the stack controller to wait for it.  

I suppose the Program's generation field is sufficient for detecting a program change.

Reply:
Since we're doing this 'on-demand', having a ready condition doesn't add much value. To keep the status block simple for now, we can use the `.status.artifact.url` field to detect when the program can be downloaded.